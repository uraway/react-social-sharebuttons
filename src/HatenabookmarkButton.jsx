import React from 'react';

export default class HatenabookmarkButton extends React.Component {
  constructor(props) {
    super(props);
    this.state = ({ initialized: false });
  }

  componentDidMount() {
    if (this.state.initialized) {
      return;
    }

    const hatenabutton = this.node;
    const hatenascript = document.createElement('script');
    hatenascript.src = 'https://b.st-hatena.com/js/bookmark_button.js';
    hatenascript.type = 'text/javascript';
    hatenabutton.parentNode.appendChild(hatenascript);

    this.initialized();
  }

  initialized() {
    this.setState({ initialized: true });
  }

  render() {
    return (
      <a
        ref={node => this.node = node}
        href={`http://b.hatena.ne.jp/entry/${this.props.url}`}
        className="hatena-bookmark-button"
        data-hatena-bookmark-title={this.props.title}
        data-hatena-bookmark-layout={this.props.layout}
        data-hatena-bookmark-lang="ja"
        title="このエントリーをはてなブックマークに追加"
      >
        <img
          src="https://b.st-hatena.com/images/entry-button/button-only@2x.png"
          alt="このエントリーをはてなブックマークに追加"
          style={{ width: 20, height: 20, border: 'none' }}
        />
      </a>
    );
  }
}

HatenabookmarkButton.propTypes = {
  url: React.PropTypes.string,
  layout: React.PropTypes.string,
  title: React.PropTypes.string,
};

HatenabookmarkButton.defaultProps = {
  url: (typeof window !== 'undefined' ? window.url : ''),
  layout: '',
  title: ''
};
