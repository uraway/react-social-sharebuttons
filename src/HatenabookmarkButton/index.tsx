import { FC, useEffect } from 'react';

declare let window: Window & { Hatena: unknown };

export type HatenaBookmarkButtonProps = {
  url?: string;
  title?: string;
  layout?: 'basic-label-counter' | 'touch-counter' | 'vertical-large';
};

const HatenaBookmarkButton: FC<HatenaBookmarkButtonProps> = (props) => {
  useEffect(() => {
    const script = document.createElement('script');
    script.src = 'https://b.st-hatena.com/js/bookmark_button.js';
    script.type = 'text/javascript';
    document.getElementsByTagName('head')[0]?.appendChild(script);
  }, []);

  const { url, title, layout, ...others } = props;

  return (
    <div>
      <a
        href={`http://b.hatena.ne.jp/entry/${url || ''}`}
        className="hatena-bookmark-button"
        data-hatena-bookmark-title={title}
        data-hatena-bookmark-layout={layout}
        data-hatena-bookmark-lang="ja"
        title="このエントリーをはてなブックマークに追加"
      >
        <img
          src="https://b.st-hatena.com/images/entry-button/button-only@2x.png"
          alt="このエントリーをはてなブックマークに追加"
          style={{ width: 20, height: 20, border: 'none' }}
          {...others}
        />
      </a>
    </div>
  );
};

HatenaBookmarkButton.defaultProps = {
  url: typeof window !== 'undefined' ? window.location.href : '',
  layout: 'basic-label-counter',
  title: '',
};

export default HatenaBookmarkButton;
