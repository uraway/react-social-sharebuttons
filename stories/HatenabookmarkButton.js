import React from 'react';
import { text, select } from '@storybook/addon-knobs';
import { HatenabookmarkButton } from '../dist/index.es.js'

export default {
  title: 'HatenabookmarkButton',
};

const layoutOptions = {
  'basic-label-counter': 'basic-label-counter',
  'touch-counter': 'touch-counter',
  'vertical-large': 'vertical-large',
};

export const Basic = () => (
  <HatenabookmarkButton
    url={text('url', '')}
    text={text('title', '')}
    layout={select('layout', layoutOptions, 'basic-label-counter')}
  />
);
