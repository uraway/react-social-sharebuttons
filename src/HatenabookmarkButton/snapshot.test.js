import React from 'react';
import renderer from 'react-test-renderer';
import HatenabookmarkButton from '.';

it('renders correctly with required props', () => {
  const tree = renderer.create(<HatenabookmarkButton />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with other props', () => {
  const tree = renderer.create(<HatenabookmarkButton url="url" title="title" layout="standard-balloon" />).toJSON();
  expect(tree).toMatchSnapshot();
});
