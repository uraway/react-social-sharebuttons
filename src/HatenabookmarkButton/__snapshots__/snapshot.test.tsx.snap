// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with other props 1`] = `
<div>
  <a
    className="hatena-bookmark-button"
    data-hatena-bookmark-lang="ja"
    data-hatena-bookmark-layout="basic-label-counter"
    data-hatena-bookmark-title="title"
    href="http://b.hatena.ne.jp/entry/url"
    title="このエントリーをはてなブックマークに追加"
  >
    <img
      alt="このエントリーをはてなブックマークに追加"
      src="https://b.st-hatena.com/images/entry-button/button-only@2x.png"
      style={
        Object {
          "border": "none",
          "height": 20,
          "width": 20,
        }
      }
    />
  </a>
</div>
`;

exports[`renders correctly with required props 1`] = `
<div>
  <a
    className="hatena-bookmark-button"
    data-hatena-bookmark-lang="ja"
    data-hatena-bookmark-layout=""
    data-hatena-bookmark-title=""
    href="http://b.hatena.ne.jp/entry/http://localhost/"
    title="このエントリーをはてなブックマークに追加"
  >
    <img
      alt="このエントリーをはてなブックマークに追加"
      src="https://b.st-hatena.com/images/entry-button/button-only@2x.png"
      style={
        Object {
          "border": "none",
          "height": 20,
          "width": 20,
        }
      }
    />
  </a>
</div>
`;
